### PR TITLE
Create OCL.g4

### PR DESCRIPTION
**Pull request by kevinlano**
Parser for OCL2.4

### CODE REVIEW

File: ocl/OCL.g4 [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+specification:
+  'package' ID '{' classifier* '}' EOF;
+
+classifier: classDefinition
+    | interfaceDefinition
+    | usecaseDefinition
+    | enumeration;
+
+interfaceDefinition:	 'interface' ID '{' classBody? '}' 
+    | 'interface' ID 'extends' ID '{' classBody? '}';
```
Comment (KvanTTT) [inline ocl/OCL.g4:27]: The common part can be extracted here and below:

```suggestion
interfaceDefinition
    : 'interface' ID ('extends' ID)? '{' classBody? '}'
    ;
```

Also, please use such formatting (we use it across all grammars in the repository):

```
rule
    : a1 (a2 | a3)
    | a4
    ...
    | an
    ;
```

File: ocl/OCL.g4 [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+classifier: classDefinition
+    | interfaceDefinition
+    | usecaseDefinition
+    | enumeration;
+
+interfaceDefinition:	 'interface' ID '{' classBody? '}' 
+    | 'interface' ID 'extends' ID '{' classBody? '}'; 
+
+classDefinition:	 'class' ID '{' classBody? '}' 
+    | 'class' ID 'extends' ID '{' classBody? '}';
```
Comment (KvanTTT) [inline ocl/OCL.g4:30]:
```suggestion
classDefinition
    : 'class' ID ('extends' ID)? '{' classBody? '}'
    ; 
```

File: ocl/OCL.g4 [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+    | operationDefinition
+    | invariant 
+    | stereotype; 
+
+attributeDefinition: 
+      'attribute' ID ':' type ';' |
+      'attribute' ID 'identity' ':' type ';' |
+      'attribute' ID 'derived' ':' type ';' | 
+      'static' 'attribute' ID ':' type ';'
+    ;
```
Comment (KvanTTT) [inline ocl/OCL.g4:44]:
```suggestion
attributeDefinition
    : 'attribute' ID ('identity' | 'derived')? ':' type ';'
    | 'static' 'attribute' ID ':' type ';'
    ; 
```

**Review by KvanTTT — CHANGES_REQUESTED** [review]
Please reformat and simplify.

**Comment by teverett** [comment]
@kevinlano thanks for this, please fix the merge conflicts.

**Comment by teverett** [comment]
@kevinlano  thanks for this.  I'll pull this PR and then format the code with:

https://github.com/antlr/Antlr4Formatter

If that works for you?

**Comment by kevinlano** [comment]
Thanks Tom,
yes, that is fine,
Kevin
________________________________

**Comment by KvanTTT** [comment]
It looks like the request already is not actual.

**Pull request closed without merging**